crates/transaction-emitter-lib/src/emitter/mod.rs

```diff
@@ -1038,7 +1058,22 @@ pub async fn create_accounts(
         max_retries: retries,
         retry_after: req.init_retry_interval,
     };
-    account_minter
-        .create_accounts(&txn_executor, req, max_submit_batch_size, num_accounts)
-        .await
+    let mut rng = StdRng::from_seed(seed);
+
+    let accounts = gen_reusable_accounts(&txn_executor, num_accounts, &mut rng).await?;
+    if !skip_minting_accounts {
+        let accounts: Vec<_> = accounts.into_iter().map(Arc::new).collect();
+        account_minter
+            .create_and_fund_accounts(&txn_executor, req, max_submit_batch_size, accounts.clone())
+            .await?;
+        let accounts: Vec<_> = accounts
+            .into_iter()
+            .map(|a| Arc::try_unwrap(a).unwrap())
+            .collect();
+        info!("Accounts created and funded");
+        Ok(accounts)
+    } else {
+        info!("Skipping minting accounts");
```

Comment:
What you have in mind? Just check for non zero balance? Or something more sophisticated?

Reply:
we compute needed balance (and that's the balance we mint to accounts). I was thinking of checking that balance?

we do similar thing for the source account itself.


Alternatively, instead of this flag, we could check if accounts exist and their balance, and always only top them up to the wanted amount

Reply:
All right added a check for needed balance.